clients/unshaded/src/main/java/tachyon/client/file/BaseFileSystem.java

```diff
@@ -0,0 +1,296 @@
+      throws FileAlreadyExistsException, InvalidPathException, IOException, TachyonException {
+    return createFile(path, CreateFileOptions.defaults());
+  }
+
+  @Override
+  public FileOutStream createFile(TachyonURI path, CreateFileOptions options)
+      throws FileAlreadyExistsException, InvalidPathException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = mContext.acquireMasterClient();
+    try {
+      masterClient.createFile(path, options);
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+    return new FileOutStream(path, options.toOutStreamOptions());
+  }
+
+  @Override
+  public void delete(TachyonURI path)
+      throws DirectoryNotEmptyException, FileDoesNotExistException, IOException, TachyonException {
+    delete(path, DeleteOptions.defaults());
+  }
+
+  @Override
+  public void delete(TachyonURI path, DeleteOptions options)
+      throws DirectoryNotEmptyException, FileDoesNotExistException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = mContext.acquireMasterClient();
+    try {
+      masterClient.delete(path, options);
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+  }
+
+  @Override
+  public boolean exists(TachyonURI path)
+      throws InvalidPathException, IOException, TachyonException {
+    return exists(path, ExistsOptions.defaults());
+  }
+
+  @Override
+  public boolean exists(TachyonURI path, ExistsOptions options)
+      throws InvalidPathException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = mContext.acquireMasterClient();
+    try {
+      // TODO(calvin): Make this more efficient
+      masterClient.getStatus(path);
+      return true;
+    } catch (FileDoesNotExistException e) {
+      return false;
+    } catch (InvalidPathException e) {
+      return false;
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+  }
+
+  @Override
+  public void free(TachyonURI path)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    free(path, FreeOptions.defaults());
+  }
+
+  @Override
+  public void free(TachyonURI path, FreeOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = mContext.acquireMasterClient();
+    try {
+      masterClient.free(path, options);
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+  }
+
+  @Override
+  public URIStatus getStatus(TachyonURI path)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    return getStatus(path, GetStatusOptions.defaults());
+  }
+
+  @Override
+  public URIStatus getStatus(TachyonURI path, GetStatusOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = mContext.acquireMasterClient();
+    // TODO(calvin): Fix the exception handling in the master
```

Comment:
I've removed the outdated TODO. Another thing we may want to look at is combining FileDoesNotExist and InvalidPath, imo InvalidPath should not require a server check and should only be thrown when the uri is malformed, but it is currently being used when the path is not applicable as well (ie. for mounting).